### DISCUSSION TITLE
Add ability to easily extend with custom componentry

### DISCUSSION THREAD

**Discussion by jeffpreussner** [shadcn-ui/ui#4763] [Ideas]
looking for the best way to easily extend Shadcn component registry either passing a registry URL or a registry:custom category? so i can extend the registry and add my own components why not breaking the upgrade path.

**Comment by fcislab** [comment]
Hey @jeffpreussner, I came here to open that same discussion!

From my research, currently, [Shadxn](https://github.com/anis-marrouchi/shadxn) is the best way to do that. But I haven't tested it out yet (I don't know if it's compatible with the latest shadcn update).

I have a feeling it might be a while until Shadcn itself releases a tool for the community as they released the remote registry feature for their V0 Ai tool.

Regardless, I can't wait for that feature. It'll change the game forever!
